.github/workflows/ci-workflow.yml

```diff
@@ -162,6 +162,10 @@ jobs:
           distribution: 'corretto'
           java-version: '21'
           cache: 'maven'
+      - name: Set up Python 3.12
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
```

Comment:
This is a bugfix: we were already using Python in unit tests, and without this step, the CI jobs were using whatever Python version was available by default in the GitHub Action image (which happened to be a homebrew-installed Python 3.13 in the case of the MacOS image).  With the explicit setup step, now the CI jobs are using Python 3.12 in a predictable and reproducible way.

Also, with system tests using Python 3.10 and unit tests using Python 3.12, we can verify that the Python components in NiFi work with both, which is a good thing.

Reply:
Thanks for clarifying, that makes sense, and I agree that installing a specific version is the right approach.